File: task2/pattern.java
public class Pattern {
    public static void main(String[] args) {
          int n=6;
          int k=10;
          pattern1(n);
          pattern2(n);
         pattern3(n);
         pattern4(n);
         pattern5(k);

    }

    private static void pattern5(int n) {
        for(int i=0;i<n;i++){
            for(int j=n-i;j>=1;j--){
                System.out.print(" ");
            }
            for(int j=0;j<=i;j++){
                System.out.print("*");
            }

            for(int k=1;k<=i;k++){
                System.out.print("*");
            }
            System.out.println("");
        }
        for(int i=n-1;i>0;i--){

            for(int j=n-i;j>=0;j--){
                System.out.print(" ");
            }
            for(int k=0;k<i;k++){
                System.out.print("*");
            }
            for(int l=i-2;l>=0;l--){
                System.out.print("*");
            }
            System.out.println(" ");

        }

    }

    private static void pattern4(int n) {
        for(int i=0;i<n;i++){
            for(int j=n-i;j>=1;j--){
                System.out.print(" ");
            }
            for(int j=0;j<=i;j++){
                System.out.print("*");
            }

           for(int k=1;k<=i;k++){
               System.out.print("*");
           }
            System.out.println();
        }
    }

    private static void pattern3(int n) {
        for(int i=0;i<n;i++){
            for(int j=n-i;j>=1;j--){
                System.out.print(" ");
            }
            for(int j=0;j<=i;j++){
                System.out.print("*");
            }

            System.out.println("");

        }
    }

    private static void pattern2(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<=i;j++){
                System.out.print("*");
            }
            System.out.println();
        }
    }

    private static void pattern1(int n) {
        for(int i=0;i<n;i++){
            for(int j=n-i;j>=1;j--){
                System.out.print("*");
            }
            System.out.println(" ");
        }
    }
}

File: task2/pattern.js


  function main(){
          let n=6;
          let k=10;
          pattern1(n);
          pattern2(n);
         pattern3(n);
         pattern4(n);
         pattern5(k);

    }

   function pattern5(n) {
        for(let i=0;i<n;i++){
            let line=""
            for(let j=n-i;j>=1;j--){
                line+=" "
               
            }
            for(let j=0;j<=i;j++){
                line+="*"
               
            }

            for(let k=1;k<=i;k++){
              line+="*"
            }
            console.log(line);
            
        }
    
        for(let i=n-1;i>0;i--){
             let line="";
            for(let j=n-i;j>=0;j--){
                line+=" "
            }
            for(let k=0;k<i;k++){
                line+="*"
            }
            for(let l=i-2;l>=0;l--){
              line+="*"
            }
            console.log(line);
            
        }

    }

    function pattern4( n) {
        for(let i=0;i<n;i++){
            let line="";
            for(let  j=n-i;j>=1;j--){
               line+=" "
            }
            for(let j=0;j<=i;j++){
               
                line+="*"
            }

           for(let k=1;k<=i;k++){
               
               line+="*";
           }
            console.log(line);
            
        }
    }

    function pattern3(n) {
        for(let i=0;i<n;i++){
            let line=""
            for(let j=n-i;j>=1;j--){
                line+=" "
            }
            for(let j=0;j<=i;j++){
                line+="*"
            }

          console.log(line);
          

        }
    }

    function pattern2(n) {
        for(let i=0;i<n;i++){
            let line=" "
            for(let j=0;j<=i;j++){
              line+="*"
            }
          console.log(line);
        }
    }

   function pattern1(n) {
        for(let i=0;i<n;i++){
            let line=""
            for(let j=n-i;j>=1;j--){
               line+="*";
            }
           console.log(line);
           
        }
    }
    main();

